src/stream.c

```diff
@@ -84,6 +84,7 @@ struct stream {
 	stream_rtcp_h *sessrtcph;    /**< Stream RTCP handler               */
 	stream_error_h *errorh;  /**< Stream error handler                  */
 	void *sess_arg;          /**< Session handlers argument             */
+	mtx_t *lock;             /**< Stream mutex locking                  */
```

Comment:
Which benefits does it have compared to a single vidsrc thread that drives the video frames through encoder, RTP packing and sending?

Reply:
The main problem is the `natpinhole_handler` (runs parallel from main thread).

Reply:
Video Codecs can produce burst packages sometimes. This needs paced sending without blocking encoding otherwise the Framerate drops. With a second PR I will replace the fixed sleep with a dynamic sleep bitrate calculation, like ffmpeg is handling this for udp streams.

Reply:
Ok. If possible the changes should not have influence on audio TX.
And the lock should not be shared between RX and TX, I think.

Reply:
I think, I can remove the `rtp_send` locking if new `natpinhole` recv only handling is ok from your side.